src/vs/editor/common/services/modelServiceImpl.ts

```diff
@@ -617,6 +617,7 @@ export class ModelServiceImpl extends Disposable implements IModelService {
 	}
 
 	private _onDidChangeLanguage(model: ITextModel, e: IModelLanguageChangedEvent): void {
+		model.removeAllDecorationsWithOwnerId(0);
```

Comment:
please add `// {{ SQL CARBON EDIT }}` to all changes under `vs` prior to merging.  I'm also unclear why this change is required.